packages/frontend/src/state/lp/hooks.ts

```diff
@@ -68,6 +69,62 @@ export const useOpenPositionDeposit = () => {
   return openPositionDeposit
 }
 
+// Close position with flashloan
+export const useClosePosition = () => {
+  const address = useAtomValue(addressAtom)
+  const controllerHelperContract = useAtomValue(controllerHelperHelperContractAtom)
+  const controllerContract = useAtomValue(controllerContractAtom)
+  const handleTransaction = useHandleTransaction()
+  const getDebtAmount = useGetDebtAmount()
+  const getVault = useGetVault()
+  const getPosition = useGetPosition()
+  const getQuote = useGetQuote()
+  const closePosition = useAppCallback(async (vaultId: number, onTxConfirmed?: () => void) => {
+    const vaultBefore = await getVault(vaultId)
+    const uniTokenId = vaultBefore?.NFTCollateralId 
+    const position = await getPosition(uniTokenId)
+
+    if (
+      !controllerContract ||
+      !controllerHelperContract ||
+      !address ||
+      !position ||
+      !vaultBefore ||
+      !vaultBefore.shortAmount
+    )
+      return
+
+      const shortAmount = fromTokenAmount(vaultBefore.shortAmount, OSQUEETH_DECIMALS)
+      const debtInEthPromise = getDebtAmount(shortAmount)
+      const limitEthPromise = getQuote(shortAmount, true)
+      const [debtInEth, limitEth] = await Promise.all([debtInEthPromise, limitEthPromise])
+      const collateralToFlashloan = debtInEth.multipliedBy(COLLAT_RATIO)
+
+    const flashloanCloseVaultLpNftParam = {
+      vaultId: vaultId,
+      tokenId: uniTokenId,
+      liquidity: position.liquidity,
+      liquidityPercentage: fromTokenAmount(1, 18).toFixed(0),
+      wPowerPerpAmountToBurn: shortAmount.toFixed(0),
+      collateralToFlashloan: collateralToFlashloan.toFixed(0),
+      collateralToWithdraw: 0,
+      limitPriceEthPerPowerPerp: limitEth,
+      amount0Min: 0,
+      amount1Min: 0,
+      poolFee: POOL_FEE,
+      burnExactRemoved: true,
```

Comment:
burnExactRemoved means that none of the oSQTH will be bought or sold to burn the amount of debt listed (and the ```wPowerPerpAmountToBurn``` is not used), so there is no need for the above calculations and many of the params included are not used.

I think this is a good flag to expose in the API (most users may want to burnExactRemoved, but not all)

Reply:
happy to discuss this and the above if any qs